### PR TITLE
Add main fields to reservation model

### PR DESCRIPTION
**Pull request by igordavydsson**
All the data about reservations should be now stored in separate
fields, instead of being pushed into 'data' field as simple json.

### CODE REVIEW

**Comment by igordavydsson** [comment]
Thanks for the tip @tuomas777 ! 🙂 
Got rid of those `null=True` on the `CharField`s. Will add other commits to this PR.

File: reservations/migrations/0003_from_data_to_main_fields.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+from decimal import Decimal, InvalidOperation
+from functools import reduce
+
+from django.db import IntegrityError, migrations, transaction
+
+CHAR_FIELDS_TO_DATA_KEY_MAP = {
+    'phone_number': "applicant.contact.mobile_phone",
+    'address': "applicant.postal.street_address",
+    'zip_code': "applicant.postal.postal_code",
+    'municipality': "applicant.postal.munacipality",
```
Comment (tuomas777) [inline reservations/migrations/0003_from_data_to_main_fields.py]: There seems to be a typo in "applicant.postal.mun**A**cipality".

File: reservations/migrations/0003_from_data_to_main_fields.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                    value = deep_get(r.data, data_address)
+                    if value:
+                        setattr(r, attr_name, value)
+
+                boat_weight = deep_get(r.data, "boat.weight")
+                if boat_weight:
+                    try:
+                        r.boat_weight = int(boat_weight)
+                    except (TypeError, ValueError):
+                        pass
```
Comment (tuomas777) [inline reservations/migrations/0003_from_data_to_main_fields.py]: Are you sure it is safe to just silently ignore all these conversion errors, not even some warning logging needed?

File: reservations/api.py [hunk 3]
```diff
@@ -1,13 +1,45 @@
...
+
     class Meta:
         model = Reservation
         fields = '__all__'
 
+    def create(self, validated_data):
+        # Django REST requires to explicitly write creation logic
+        # for ManyToMany relations with a Through model.
+        chosen_harbors = validated_data.pop('harborchoice_set')
+        reservation = super(ReservationSerializer, self).create(validated_data)
```
Comment (tuomas777) [inline reservations/api.py]: Just `super().create(...)` is enough in Python 3.

File: reservations/migrations/0003_from_data_to_main_fields.py (hunk 1 above)
Comment (igordavydsson) [reply reservations/migrations/0003_from_data_to_main_fields.py]: yeah, frontend had a typo there, so all the reservations now have this typo too 🙂

File: reservations/api.py (hunk 3 above)
Comment (igordavydsson) [reply reservations/api.py]: ah, right! forgot that those could be ommited. 👍

File: reservations/migrations/0003_from_data_to_main_fields.py (hunk 2 above)
Comment (igordavydsson) [reply reservations/migrations/0003_from_data_to_main_fields.py]: good point! best to add at least logger 👍

**Pull request merged**